.github/workflows/gma.lua

```diff
@@ -0,0 +1,138 @@
+-- GMA packer by Vurv
+assert(arg[1], "Missing argument #1 (output file)")
+
+local function u8(n --[[@param n number]])
+	return string.char(n)
+end
+
+local U8_MAX = 256
+local U16_MAX = 65536
+local U32_MAX = 4294967296
+
+local function u16(n --[[@param n number]])
+	return u8(n % U8_MAX) .. u8( math.floor(n / U8_MAX) )
+end
+
+local function u32(n --[[@param n number]])
+	return u16(n % U16_MAX) .. u16( math.floor(n / U16_MAX) )
+end
+
+local function u64(n --[[@param n number]])
+	return u32(n % U32_MAX) .. u32( math.floor(n / U32_MAX) )
+end
```

Comment:
Can we use lua 5.3 `pack` instead?

Reply:
Would be an even smaller implementation yeah, less readable though:

```lua
---@param name string
---@param description string
---@param author string
---@param files { path: string, content: string }[] # List of 'files'
---@return string gma # Packed gma file contents
local function pack(name, desc, author, files)
    return "GMAD"
        .. string.pack("< I1 I8 I8 I1 z z z I4", 3, 0, os.time(), 0, name, desc, author, 1)
        .. table.concat(map(files, function(v, k)
            return string.pack("< I4 z I8 I4", k, v.path, #v.content, 0)
        end))
        .. "\0\0\0\0"
        .. table.concat(map(files, function(v, k)
            return v.content
        end))
        .. "\0\0\0\0"
end
```

Fine with this?

Reply:
I think so, just test and make sure it produces the same file output